org-roam.el

```diff
@@ -821,6 +851,321 @@ for Org-ref cite links."
+    (format "roam:%s" (completing-read "Roam note: "
+                                       (-union (map-keys completions) in-buffer-completions)))))
+
+(defun org-roam--roam-link-message ()
+  "Send roam-link status message to minibuffer."
+  (when (string= major-mode "org-mode")
+    (let ((elem (org-element-context)))
+      (when (and (eq (car elem) 'link)
+                 (string= "roam" (org-element-property :type elem)))
+        (save-match-data
+          (let* ((title (org-element-property :path elem))
+                 (abs-file-path (org-roam--get-file-from-title title))
+                 (rel-file-path
+                  (when abs-file-path
+                    (string-match (format "%s\\(.*\\)" org-roam-directory) abs-file-path)))
+                 (file-path (when rel-file-path
+                              (concat "~org-roam-dir~" (match-string 1 abs-file-path))))
+                 (raw-link (org-element-property :raw-link elem)))
+            (if file-path
+                (message "file: %s → %s" file-path raw-link)
+              (message "No file found in db → %s" raw-link))))))))
+
+
+(defun org-roam-show-link-messages ()
+  "Enable minibuffer status message for roam-links.
+Follows example of `org-ref' and displays on idle timer."
+  (interactive)
+  (or org-roam-link-message-timer
+      (setq org-roam-link-message-timer
+            (run-with-idle-timer 0.5 t #'org-roam--roam-link-message))))
+
+(defun org-roam-cancel-link-messages ()
+  "Disable minibuffer status message for roam-links."
+  (interactive)
+  (cancel-timer org-roam-link-message-timer)
+  (setq org-roam-link-message-timer nil))
+
+;; Turn roam-link status message on by default
+(org-roam-show-link-messages)
+
+(defun org-roam-insert-roam-link ()
+  "Shortcut to insert roam-link with standard completion prompt."
+  (interactive)
+  (insert "[[" (org-roam--roam-link-completion) "]]"))
+
+(defun org-roam-insert-roam-syntax ()
+  "Shortcut to insert roam-link syntax without completion.
+Move point inside brackets, ready for roam title entry."
+  (interactive)
+  (insert "[[roam:]]")
+  (backward-char 2))
+
+(defun org-roam--exit-roam-link ()
+  "When point is in roam-link, advance point forward out of the link."
+  (interactive)
+  (goto-char (org-element-property :end (org-element-context))))
+
+(org-link-set-parameters
+ "roam"
+ :display 'full
+ :face 'org-roam--custom-roam-link-face
+ :activate-func 'org-roam--roam-link-activate
+ :follow 'org-roam--roam-link-find-file
+ :complete 'org-roam--roam-link-completion
+ :keymap (let ((map (copy-keymap org-mouse-map)))
+           (define-key map (kbd "M-f") 'org-roam-convert-roam-to-file-link)
+           (define-key map (kbd "<tab>") 'org-roam--exit-roam-link)
+           map))
+
+(org-link-set-parameters
+ "file"
+ :keymap (let ((map (copy-keymap org-mouse-map)))
+           (define-key map (kbd "M-r") 'org-roam-convert-file-to-roam-link)
+           map))
```

Comment:
We should move this such that this is activated within `org-roam-mode`, and reset when the mode is deactivated, rather than make a lasting change when requiring org-roam.

Reply:
Moved everything into `org-roam-mode`, but I don't know if I'm not understanding how mode activation/de-activation works. On my Doom setup everything loads like I would expect even in just a normal org file. On manual de-activation the settings are reset as expected, but manaully re-enabling `org-roam-mode` DOES NOT re-activate `roam:` links with all of its faces and functions.